### PR TITLE
POV-437

### PR DESCRIPTION
**Pull request by marinhekman**
Export a region of the msp data either from console or api 

* For the api, post to /{sessionId}/api/Game/CreatePOVConfig will return a zip response containing the data

image

* The console can output a directory or a zip file ,see below. Easy usage will be `php bin/console app:create-pov-config 1 4048536 3470232 4063746 3488599 -v`
The -v verbose option is advised to see the status notifications
```
/srv/app # php bin/console app:create-pov-config --help
Description:
  Create a POV config json file for a session by region.

Usage:
  app:create-pov-config [options] [--] <session-id> <region-coordinates>...

Arguments:
  session-id                                             The ID of the game session
  region-coordinates                                     4 floats representing the coordinates of the region. In order: region top left coordinates x, -y, region bottom right coordinates x, -y. Eg: 4048536 3470232 4063746 3488599

Options:
  -d, --output-dir=OUTPUT-DIR                            The path to output directory. Default is: /srv/app/POV
  -j, --output-json-filename=OUTPUT-JSON-FILENAME        The filename of output json file. Default is: pov-config.json [default: "pov-config.json"]
  -p, --output-package-filename=OUTPUT-PACKAGE-FILENAME  The filename of compressed package file if compression is enabled. Default is based on the region coordinates like: 4048536-3470232-4063746-3488599.zip
  -z, --compress                                         Enables zip compression of the output to a package file.
  -h, --help                                             Display help for the given command. When no command is given display help for the list command
  -q, --quiet                                            Do not output any message
  -V, --version                                          Display this application version
      --ansi|--no-ansi                                   Force (or disable --no-ansi) ANSI output
  -n, --no-interaction                                   Do not ask any interactive question
  -e, --env=ENV                                          The Environment name. [default: "dev"]
      --no-debug                                         Switch off debug mode.
  -v|vv|vvv, --verbose                                   Increase the verbosity of messages: 1 for normal output, 2 for more verbose output and 3 for debug
```

Finally, to be able to do "bash install.sh" from git bash you need a php extension imagick.
@hwarmelink I advise you do the same as me . You remove Xampp entirely and go for 
https://github.com/mlocati/powershell-phpmanager
It is a php manager from powershell (with admin rights). It works like this:

* Install it: `Install-Module -Name PhpManager -Repository PSGallery -Force`
* Install php 8.1 (that the requirement of MSP): `Install-Php -Version 8.1 -Architecture x64 -ThreadSafe 0 -Path C:\PHP -TimeZone UTC -AddToPath User`
* Check the version(s) installed (you can have multiple): `Get-Php C:\PHP`
* If you have multiple, check the doc: https://github.com/mlocati/powershell-phpmanager#working-with-multiple-php-installations
* Enable required openssl php extension: `Enable-PhpExtension openssl C:\PHP`
* Enable required curl php extension: `Enable-PhpExtension curl C:\PHP`
* Enable required intl php extension: `Enable-PhpExtension intl C:\PHP`
* Enable required sodium php extension: `Enable-PhpExtension sodium C:\PHP`
* Install imagick PECL php extension: `Install-PhpExtension imagick -MinimumStability snapshot`

I do not think you need xampp or xdebug, since we use docker for that now 

Otherwise, you could go download imagick extension for xampp (so a compiled dll binary) from https://mlocati.github.io/articles/php-windows-imagick.html. Add is to the php's ext folder etc.

### CODE REVIEW

File: composer-symfony5.4.json [hunk 1]
```diff
@@ -8,6 +8,7 @@
         "ext-ctype": "*",
         "ext-curl": "*",
         "ext-iconv": "*",
+        "ext-imagick": "*",
```
Comment (marinhekman) [inline composer-symfony5.4.json:11]: imagick has tif support

File: docker/php/docker-entrypoint.sh [hunk 2]
```diff
@@ -66,6 +66,8 @@ if [ "$1" = 'php-fpm' ] || [ "$1" = 'php' ] || [ "$1" = 'bin/console' ]; then
 	setfacl -dR -m u:www-data:rwX -m u:"$(whoami)":rwX session_archive
 	setfacl -R -m u:www-data:rwX -m u:"$(whoami)":rwX var
 	setfacl -dR -m u:www-data:rwX -m u:"$(whoami)":rwX var
+	setfacl -R -m u:www-data:rwX -m u:"$(whoami)":rwX POV
```
Comment (marinhekman) [inline docker/php/docker-entrypoint.sh:69]: POV is the default output folder

File: migrations/Version20231204154654.php [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+use Doctrine\DBAL\Schema\Schema;
+
+/**
+ * Auto-generated Migration: Please modify to your needs!
+ */
+final class Version20231204154654 extends MSPMigration
+{
+    public function getDescription(): string
+    {
+        return 'Add index to geometry_subtractive column in geometry table';
```
Comment (marinhekman) [inline migrations/Version20231204154654.php:16]: this speeds up the configcreator query by almost 20 sec

File: src/Domain/POV/ConfigCreator.php [hunk 4]
```diff
@@ -0,0 +1,554 @@
...
+              )
+            )
+          )        
+        ) AS layer_data
+      FROM LayerStep2 l
+      # left join since raster layers do not ever have geometry data
+      LEFT JOIN LatestGeometryInRegion as g ON g.geometry_layer_id=l.layer_id
+      WHERE l.layer_geotype IN ('raster') OR g.geometry_layer_id IS NOT NULL
+      GROUP BY l.layer_id
+  )
```
Comment (marinhekman) [inline src/Domain/POV/ConfigCreator.php:285]: so the WITH part are all prepared temporary table to be used by the actual main SELECT below

Comment (hwarmelink) [reply src/Domain/POV/ConfigCreator.php:285]: wow!

**Review by hwarmelink — APPROVED** [review]

**Pull request merged**